rmw_dds_common/test/test_graph_cache.cpp

```diff
@@ -1127,3 +1128,259 @@ TEST(test_graph_cache, normal_usage)
+    rcutils_reset_error();
+  }
+
+  {
+    rcutils_allocator_t allocator = rcutils_get_default_allocator();
+    rcutils_string_array_t names = rcutils_get_zero_initialized_string_array();
+    rmw_ret_t ret = rcutils_string_array_init(&names, 3, &allocator);
+    EXPECT_EQ(ret, RMW_RET_OK);
+    rcutils_string_array_t namespaces = rcutils_get_zero_initialized_string_array();
+    ret = graph_cache.get_node_names(&names, &namespaces, nullptr, &allocator);
+    EXPECT_EQ(ret, RMW_RET_INVALID_ARGUMENT);
+    rcutils_reset_error();
+  }
+
+  {
+    rcutils_allocator_t allocator = rcutils_get_default_allocator();
+    rcutils_string_array_t names = rcutils_get_zero_initialized_string_array();
+    rcutils_string_array_t namespaces = rcutils_get_zero_initialized_string_array();
+    rmw_ret_t ret = rcutils_string_array_init(&namespaces, 3, &allocator);
+    EXPECT_EQ(ret, RMW_RET_OK);
+    ret = graph_cache.get_node_names(&names, &namespaces, nullptr, &allocator);
+    EXPECT_EQ(ret, RMW_RET_INVALID_ARGUMENT);
+    rcutils_reset_error();
+  }
+
+  {
+    rcutils_allocator_t allocator = rcutils_get_default_allocator();
+    rcutils_string_array_t names = rcutils_get_zero_initialized_string_array();
+    rcutils_string_array_t namespaces = rcutils_get_zero_initialized_string_array();
+    rcutils_string_array_t enclaves = rcutils_get_zero_initialized_string_array();
+    rmw_ret_t ret = rcutils_string_array_init(&enclaves, 3, &allocator);
+    EXPECT_EQ(ret, RMW_RET_OK);
+    ret = graph_cache.get_node_names(&names, &namespaces, &enclaves, &allocator);
+    EXPECT_EQ(ret, RMW_RET_INVALID_ARGUMENT);
+    rcutils_reset_error();
+  }
+
+  {
+    rcutils_allocator_t allocator = rcutils_get_default_allocator();
+    rmw_names_and_types_t names_and_types = rmw_get_zero_initialized_names_and_types();
+    rmw_ret_t ret = graph_cache.get_writer_names_and_types_by_node(
+      "node_name",
+      "node_namespace",
+      identity_demangle,
+      identity_demangle,
+      &allocator,
+      &names_and_types);
+    EXPECT_EQ(ret, RMW_RET_NODE_NAME_NON_EXISTENT);
+    rcutils_reset_error();
+    rcutils_allocator_t zero_allocator = rcutils_get_zero_initialized_allocator();
+    ret = graph_cache.get_writer_names_and_types_by_node(
+      "node_name",
+      "node_namespace",
+      identity_demangle,
+      identity_demangle,
+      &zero_allocator,
+      &names_and_types);
+    EXPECT_EQ(ret, RMW_RET_INVALID_ARGUMENT);
+    rcutils_reset_error();
+  }
+
+  // Add one participant.
+  add_participants(graph_cache, {"participant1"});
+  {
+    rmw_topic_endpoint_info_array_t topic_endpoint_info_array_sub =
+      rmw_get_zero_initialized_topic_endpoint_info_array();
+    rcutils_allocator_t allocator = rcutils_get_default_allocator();
+    rmw_ret_t ret = graph_cache.get_writers_info_by_topic(
+      "topic1",
+      identity_demangle,
+      &allocator,
+      &topic_endpoint_info_array_sub);
+    EXPECT_EQ(ret, RMW_RET_OK);
+  }
```

Comment:
Would it be possible/reasonable to check here the output of  the function, written into `topic_endpoint_info_array_sub`?

Reply:
https://github.com/ros2/rmw_dds_common/pull/14/commits/9f4a0f5269cc7a5e6e85f1356d236d00cd60c5c9